### PR TITLE
Completed the Project 

### PR DESCRIPTION
**Pull request by AsimKhan2019**
Completed all the requirements.

Implemented the requirements for missions.
Implemented the requirements for rockets
Implemented the tests.

A special thanks to my coding Partner Bien Venu Ushindi

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    "prop-types": "^15.8.1",
+    "react": "^18.2.0",
+    "react-bootstrap": "^2.5.0",
+    "react-dom": "^18.2.0",
+    "react-redux": "^8.0.2",
+    "react-router-dom": "^6.3.0",
+    "react-scripts": "5.0.1",
+    "react-test-renderer": "^18.2.0",
+    "redux-logger": "^3.0.6",
+    "redux-thunk": "^2.4.1",
```
Comment (mateo951) [inline package.json:19]: [React Redux](https://react-redux.js.org/), [Redux Logger](https://www.npmjs.com/package/redux-logger) and [React Router](https://reactrouter.com/web/guides/quick-start) exist on the package.json file ✔

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+function App() {
+  return (
+    <div className="App">
+      <Nav />
+      <Routes>
+        <Route exact path="/" element={<Rockets />} />
+        <Route path="/missions" element={<Mission />} />
+        <Route path="/profile" element={<Profile />} />
+      </Routes>
```
Comment (mateo951) [inline src/App.js:17]: Proper routes and view components are created properly 🙌

File: src/index.css [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  text-decoration: none;
+  background-color: transparent;
+}
+
+nav ul a:focus {
+  text-decoration: underline;
+}
+
+nav ul a:active {
+  text-decoration: underline;
```
Comment (mateo951) [inline src/index.css:74]: The active page should be indicated with an underline ✔

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import { BrowserRouter } from 'react-router-dom';
+import spaceStore from './Components/Redux/spaceStore';
+import App from './App';
+import './index.css';
+
+const root = ReactDOM.createRoot(document.getElementById('root'));
+
+root.render(
+  <React.StrictMode>
+    <Provider store={spaceStore}>
```
Comment (mateo951) [inline src/index.js:13]: Redux store is configured properly! Well done 🔥

File: src/Components/Redux/Mission/missionAPI.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+/* eslint-disable camelcase */
+
+const url = 'https://api.spacexdata.com/v3/missions';
```
Comment (mateo951) [inline src/Components/Redux/Mission/missionAPI.js:3]: Rockets and Missions data is being fetched successfully from SpaceX API 🤓

File: src/Components/Mission/Mission.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+import { useDispatch, useSelector } from 'react-redux';
+import MissionList from './missionList';
+import { getMissions, getJoined, switchMissions } from '../Redux/Mission/Actions/missionAction';
+
+const Mission = () => {
+  const dispatch = useDispatch();
+  const missions = useSelector((state) => state.missions);
+  useEffect(() => {
+    if (missions.length === 0) dispatch(getMissions());
+    else dispatch(getJoined(missions));
```
Comment (mateo951) [inline src/Components/Mission/Mission.js:11]: "Join Mission" button updates the store successfully by dispatching an action 💪😎 Well done ✔

File: src/Components/Profile/Profile.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        <ul className="list-group list-group-flush">
+          {list}
+        </ul>
+      </div>
+      <div className="col-md-6 card">
+        <h2>My Reserved Rockets</h2>
+        <ul className="list-group list-group-flush">
+          {rocketsList}
+        </ul>
+      </div>
```
Comment (mateo951) [inline src/Components/Profile/Profile.js:25]: Awesome 🔥 You are rendering a list of reserved rockets and missions without any issue!

File: src/Components/Profile/Profile.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
+/* eslint-disable camelcase */
+
+import { useSelector } from 'react-redux';
+
+const Profile = () => {
+  const missions = useSelector((state) => state.missions);
+  const rockets = useSelector((state) => state.rockets);
+  const rocketsList = rockets.filter((r) => r.reserved === true).map((rocket) => (<li key={rocket.id} className="list-group-item">{rocket.name}</li>));
+  const list = missions.filter((r) => r.joined === true).map((mission) => (
```
Comment (mateo951) [inline src/Components/Profile/Profile.js:9]: The `filter()` method is used to render the reserved rockets and missions 🚀 Fantastic 🙌

**Review by mateo951 — APPROVED** [review]
### Hi Team, 

📝👌 **Project highlights** 😎 
- General Requirements 💪🤓
- Gitflow (Reviews for pair-programming are good) 🙌
- Data fetch from API's 🤯
- Rendering components properly 🚀

You did a fabulous job 😁 You should be very proud of yourselves 😎

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

File: package.json (hunk 1 above)
Comment (AsimKhan2019) [reply package.json:19]: @mateo951 Thank you for the approval

File: src/App.js (hunk 2 above)
Comment (AsimKhan2019) [reply src/App.js:17]: @mateo951 Thank you for the approval

File: src/index.css (hunk 3 above)
Comment (AsimKhan2019) [reply src/index.css:74]: @mateo951 Thank you for the approval

File: src/index.js (hunk 4 above)
Comment (AsimKhan2019) [reply src/index.js:13]: @mateo951 Thank you for the approval

File: src/Components/Redux/Mission/missionAPI.js (hunk 5 above)
Comment (AsimKhan2019) [reply src/Components/Redux/Mission/missionAPI.js:3]: @mateo951 Thank you for the approval

File: src/Components/Mission/Mission.js (hunk 6 above)
Comment (AsimKhan2019) [reply src/Components/Mission/Mission.js:11]: @mateo951 Thank you for the approval

File: src/Components/Profile/Profile.js (hunk 7 above)
Comment (AsimKhan2019) [reply src/Components/Profile/Profile.js:25]: @mateo951 Thank you for the approval

File: src/Components/Profile/Profile.js (hunk 8 above)
Comment (AsimKhan2019) [reply src/Components/Profile/Profile.js:9]: @mateo951 Thank you for the approval

**Pull request merged**